ldms/src/core/ldms_xprt.c

```diff
@@ -347,34 +347,42 @@ static void send_dir_update(struct ldms_xprt *x,
 {
 	size_t hdr_len;
 	size_t buf_len;
-	size_t cnt;
+	size_t json_data_len;
 	struct ldms_reply *reply;
 
 	assert(t != LDMS_DIR_LIST);
 
 	hdr_len = sizeof(struct ldms_reply_hdr)
 		+ sizeof(struct ldms_dir_reply);
-	buf_len = ldms_xprt_msg_max(x);
-	reply = malloc(buf_len);
+	/* Calculate the size of the string */
+	json_data_len = snprintf(NULL, 0, "{ \"directory\" : [ %s ]}", json);
+	buf_len = hdr_len + json_data_len;
 
-	cnt = snprintf(reply->dir.json_data, buf_len - hdr_len,
-		       "{ \"directory\" : [ %s ]}", json);
-	if (cnt >= buf_len)
+	if (buf_len >= ldms_xprt_msg_max(x))
 		goto out;
+
+	reply = malloc(buf_len);
+	if (!reply) {
+		if (x->log)
+			x->log("%s: Out of memory\n", __FUNCTION__);
+		return;
+	}
+
 	reply->hdr.xid = x->remote_dir_xid;
 	reply->hdr.cmd = htonl(LDMS_CMD_DIR_UPDATE_REPLY);
 	reply->hdr.rc = 0;
 	reply->dir.type = htonl(t);
-	reply->dir.json_data_len = htonl(cnt);
```

Comment:
Let's set it to -1 to indicate that it is deprecated...

Reply:
I added the line back so that the path in 4.3.4 is compatible with 4.3.3. __process_dir_reply() in v4.3.3 still uses reply->dir.json_data_len.

Reply:
Does 4.3.3 use it correctly as ntohl(??) or is it expecting the unswapped value?

Reply:
Also, I don't see where this line has been added back.